actions/trigger-argo-workflow/cmd/trigger-argo-workflow/main.go

```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	"github.com/lmittmann/tint"
+	cli "github.com/urfave/cli/v2"
+	"github.com/willabides/actionslog"
+	"github.com/willabides/actionslog/human"
+	"golang.org/x/term"
+)
+
+const (
+	flagAddCILabels      = "add-ci-labels"
+	flagArgoToken        = "argo-token"
+	flagInstance         = "instance"
+	flagLogLevel         = "log-level"
+	flagNamespace        = "namespace"
+	flagParameter        = "parameter"
+	flagRetries          = "retries"
+	flagWorkflowTemplate = "workflow-template"
+)
+
+func parseLogLevel(level string) (slog.Level, error) {
+	switch level {
+	case "debug":
+		return slog.LevelDebug, nil
+	case "info":
+		return slog.LevelInfo, nil
+	case "warn":
+		return slog.LevelWarn, nil
+	case "error":
+		return slog.LevelError, nil
+	default:
+		return slog.LevelInfo, fmt.Errorf("invalid log level: `%s`. choose from: debug, info, warn, error", level)
+	}
+}
+
+func main() {
+	// If we're on a terminal, we use tint, otherwise if we're on GitHub actions
+	// we use `willabites/actionslog` to log proepr Actions messages, otherwise
+	// we use logfmt.
+	var lv slog.LevelVar
+
+	logger := slog.New(
+		slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{
+			Level: &lv,
+		}),
+	)
+
+	if os.Getenv("GITHUB_ACTIONS") == "true" {
+		handler := &human.Handler{Level: &lv}
+
+		logger = slog.New(&actionslog.Wrapper{
+			Handler: handler.WithOutput,
+		})
+	}
+
+	if term.IsTerminal(int(os.Stderr.Fd())) {
+		logger = slog.New(
+			tint.NewHandler(os.Stderr, &tint.Options{
+				Level: &lv,
+			}),
+		)
+	}
+
+	app := cli.NewApp()
+	app.Name = "Runs the Argo CLI"
+
+	app.Action = func(c *cli.Context) error { return run(c, &lv, logger) }
+
+	app.Flags = []cli.Flag{
+		&cli.BoolFlag{
+			Name:    flagAddCILabels,
+			EnvVars: []string{"ADD_CI_LABELS"},
+			Value:   false,
+			Usage:   "If true, the `--labels` argument will be added with values from the environment. This is forced for the `submit` command",
+		},
+		&cli.StringFlag{
+			Name:     flagNamespace,
+			EnvVars:  []string{"ARGO_NAMESPACE"},
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:     flagArgoToken,
+			EnvVars:  []string{"ARGO_TOKEN"},
+			Usage:    "The Argo token to use for authentication",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:    flagLogLevel,
+			EnvVars: []string{"LOG_LEVEL"},
+			Usage:   "Which log level to use",
+			Value:   "info",
+			Action: func(c *cli.Context, level string) error {
+				level = strings.ToLower(level)
+
+				lvl, err := parseLogLevel(level)
+				if err != nil {
+					return err
+				}
+
+				lv.Set(lvl)
+
+				return nil
+			},
+		},
+		&cli.StringFlag{
+			Name:    flagInstance,
+			EnvVars: []string{"INSTANCE"},
+			Value:   "ops",
```

Comment:
Hm, shouldn't be safer if the default was `dev` instead?

Reply:
It's about the ease of use. The normal case will be for real CD workloads and I don't want to have a parameter which everybody ends up setting to the same non-default value. I want these actions to be as easy to use as possible.

I'll close this thread, but feel free to push back and reopen if you want!